File: ps/game-of-life.cpp
class Solution {
public:
    void gameOfLife(vector<vector<int>>& board) {
        int i,j;
        for (i=0;i<board.size();i++){
            for(j=0;j<board[0].size();j++) {
                int live = board[i][j] & 1;
                int neighbor = 0;
                if (i != 0 && j != 0) neighbor += board[i-1][j-1] & 1;
                if (i != board.size()-1 && j != board[0].size()-1) 
                    neighbor += board[i+1][j+1] & 1;
                if (i != 0 && j != board[0].size()-1)
                    neighbor += board[i-1][j+1] & 1;
                if (i != board.size()-1 && j != 0)
                    neighbor += board[i+1][j-1] & 1;
                if (i != 0) neighbor += board[i-1][j] & 1;
                if (j != 0) neighbor += board[i][j-1] & 1;
                if (i != board.size()-1) neighbor += board[i+1][j] & 1;
                if (j != board[0].size()-1) neighbor += board[i][j+1] & 1;
                
                if (live && neighbor < 2) continue;
                if (live && (neighbor == 2 || neighbor == 3 )) board[i][j] |= 2;
                if (live && neighbor > 3) continue;
                if (!live && neighbor == 3) board[i][j] |= 2;
            }
        }
        for (i=0;i<board.size();i++){
            for(j=0;j<board[0].size();j++) {
                board[i][j] >>= 1;
            }
        }
        
    }
};

File: ps/game-of-life.py
class Solution:
    def gameOfLife(self, board: List[List[int]]) -> None:
        """
        Do not return anything, modify board in-place instead.
        """
        prev_list = []
        for i in range(len(board)):
            curr_list = list(board[i])

            for j in range(len(board[i])):
                numOfLivingNeighbor = 0
                if i != 0:
                    numOfLivingNeighbor += prev_list[j]
                if j != 0:
                    numOfLivingNeighbor += curr_list[j - 1]
                if i != 0 and j != 0:
                    numOfLivingNeighbor += prev_list[j - 1]

                if i != len(board) - 1:
                    numOfLivingNeighbor += board[i + 1][j]
                if j != len(board[i]) - 1:
                    numOfLivingNeighbor += curr_list[j + 1]
                if i != len(board) - 1 and j != len(board[i]) - 1:
                    numOfLivingNeighbor += board[i + 1][j + 1]

                if i != 0 and j != len(board[i]) - 1:
                    numOfLivingNeighbor += prev_list[j + 1]
                if j != 0 and i != len(board) - 1:
                    numOfLivingNeighbor += board[i + 1][j - 1]

                if board[i][j] == 1:
                    if numOfLivingNeighbor < 2 or numOfLivingNeighbor > 3:
                        board[i][j] = 0
                else:
                    if numOfLivingNeighbor == 3:
                        board[i][j] = 1

            prev_list = curr_list

